webapp/sources/rudder/rudder-core/src/main/scala/com/normation/cfclerk/xmlparsers/TechniqueParser.scala

```diff
@@ -106,7 +106,16 @@ class TechniqueParser(
 
               parseAgentConfig(id, forCompatibilityAgent)
             }
-          )
+          // we need to filter back for totally empty agent (most likely default one added for nothing)
+          ).filter(a => List(a.templates, a.files, a.bundlesequence, a.runHooks).exists(_.nonEmpty))
```

Comment:
We could use .forall(_.empty) instead of .exists(_.nonEmpty) which seems easier to understand, or maybe just changing the comment to match the code because  i exepected something else